helpers/remote.py

```diff
@@ -1,100 +1,181 @@
 """
-Controls node over SSH if remote, or via OS if local one.
-The API is required to transparently handle both the cases - Tempesta and the
-framework on the same node (developer tests case) or on separate machines (CI
-case).
+Module to work with nodes: local and remote(via SSH).
+
+The API is required to transparently handle both cases - Tempesta and the framework
+on the same node (developer test case), or on separate machines (CI case).
 """
 from __future__ import print_function
 
 import abc
 import errno
+import logging
 import os
 import re
-import shutil
 import subprocess
 import time
-from dataclasses import dataclass
-from typing import Union  # TODO: use | instead when we move to python3.10
+from typing import Any, Optional
 
 import paramiko
 
 import run_config
 
-from . import error, tf_cfg
+from helpers import error, tf_cfg
 
 __author__ = "Tempesta Technologies, Inc."
-__copyright__ = "Copyright (C) 2017 Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2024 Tempesta Technologies, Inc."
 __license__ = "GPL2"
 
+LOGGER = logging.getLogger(__name__)
+
 # Don't remove files from remote node. Helpful for tests development.
 DEBUG_FILES = False
 # Default timeout for SSH sessions and command processing.
 DEFAULT_TIMEOUT = 10
 
 
-class Node(object, metaclass=abc.ABCMeta):
-    def __init__(self, type, hostname, workdir):
+class INode(object, metaclass=abc.ABCMeta):
+    """Node interfaces."""
+
+    LOGGER = LOGGER
+
+    DEBUG_FILES = DEBUG_FILES
```

Comment:
I like such approach: I do not like to use components from outside of class + it simplify a coding process).
There is no harm from such approach.

We may call it composition (linking an object to a class) as OOP concept.

Any bad scenarios provided?

Reply:
I don't understand why create an extra nesting (link)? Why do we need global variables (LOGGER, DEBUG_FILES)? I don't think that creating unnecessary variables and references to them can be considered OOP conept. 

This is also one of the reasons why I don't want to add a logger in this PR now. I think every node (or other service) should have a personal logger or display information about where the command was run.  
For example:
`12:16:19:137211 DEBUG | ServerNode | Run command 'sysctl -w net.ipv4.tcp_tw_reuse=1' on host 'localhost' with environment '{}' remote.py:84` in this case, every node object must have a personal logger.
or other common ideas.

It seems to me that this will just add сonfusion to the code.

Reply:
We discussed everything over the phone, changes will be applied with the next commit.